### ISSUE TITLE
Kernel selection

### ISSUE THREAD

**Issue by sam3012** [microsoft/vscode-jupyter#13477]
### Applies To

- [X] Notebooks (.ipynb files)
- [X] Interactive Window and\/or Cell Scripts (.py files with \#%% markers)

### What happened?

Recently extension is not recognizing virtual environment's kernel and selecting the global one automatically. It's not even showing the virtual environment kernel in selection list. I'm using "python -m venv My_Venv"

### VS Code Version

1.78.0

### Jupyter Extension Version

v2023.4.1011241018

### Coding Language and Runtime Version

Python 3.11.3

### Language Extension Version (if applicable)

v2023.8.0

### Running Jupyter locally or remotely?

Local

**Comment by DonJayamanne** [comment]
Thanks for filing this issue and sorry you are running into this.

> It's not even showing the virtual environment kernel in selection list.

This sounds like the cause for the issue, though I'm not sure why this would happen.

Please could you enable logging as follows:
* Open settings & go into `Jupyter->Logging`
* Change the value to `verbose`
* Reload VS Code,
* Attempt to repro the issue & then copy the output from the `Jupyter` output panel.

**Comment by sam3012** [comment]
There is no such error it's just not detecting the kernel. Look images 
[Images](https://ufile.io/f/m4zmb)

**Comment by DonJayamanne** [comment]
For some reason I'm unable to download the files, please could you upload the images to github issue here, 
all you need to do is just drag an image file to a github issue here.

> There is no such error it's just not detecting the kernel. Look images

I understand there might not be any errors, however the logs would help us determine what is going on
E.g. whether you have an old version of Python extension, whether tahts installed or not,
or whether you do not have a workspce folder opened, and the like.

> Recently extension is not recognizing virtual environment's kernel and selecting the global one automatically.

Does installing a previous version of the Jupyter extension and Python extension work?

Please do share the verbose logs after enabling logging as follows
* Open settings & go into `Jupyter->Logging`
* Change the value to `verbose`
* Reload VS Code,
* Attempt to repro the issue & then copy the output from the `Jupyter` output panel.

**Comment by DonJayamanne** [comment]
I have managed to open an image, and looks like you have opened a Python environment directory, please confirm that the `ML_MATERIAL` folder is your python environment folder and this is what you have opened in VS Code.
I.e. the python executable is found here `.../ML_MATERIAL/scripts/python.exe` 

If not, please do share the logs

**Comment by sam3012** [comment]
Yes "python.exe" is there in "ML_MATERIAL/scripts/" folder and for confirmig here is the image of newly created environment..
[image](https://ufile.io/e714r85j)

**Comment by DonJayamanne** [comment]
@sam3012 thanks for the information
1. Did this work in the previous version of the jupyter & Python extension
2. Please can you share the logs, 

I'm asking for the logs to determine the cause of the issue, i.e. its likely the problem lies with Python extension and not jupyter at all.
But I'm very puzzled by the statement `Recently extension is not recognizing virtual environment's kernel `

This implies this used to work, please could you confirm this works when reverting to an older version of the Jupyter & Python extensions.

@karrtikr Does Python extension support discovering python environments when opening the Python environment folder?

**Comment by karrtikr** [comment]
No I don't think so.

@sam3012 Can you try disabling just the Jupyter extension, reload window, use 'Python: Select Interpreter' command and verify if you see the interpreter?

This would confirm whether the issue is with Python ext or Jupyter.

**Comment by DonJayamanne** [comment]
@sam3012 

rather than clicking the kernel picker please can you try clicking the `Python Select Interpreter` as follows:
* Open the command palette
* Select the command `Python: Select Interpreter` (see below)
<img width="509" alt="Screenshot 2023-05-11 at 00 38 38" src="https://github.com/microsoft/vscode-jupyter/assets/1948812/5148205a-080d-4a1e-bd05-6b4af81c72a8">

* Then take a screenshot of the Python environments displayed (see below for sample)
<img width="509" alt="Screenshot 2023-05-11 at 00 39 49" src="https://github.com/microsoft/vscode-jupyter/assets/1948812/46daa841-a865-4f9f-b690-bb0a1906e0d6">

**Comment by sam3012** [comment]
Still same.
[Images](https://ufile.io/f/4cext)

**Comment by DonJayamanne** [comment]
Confirmed, this is a bug in python extension
